### PR TITLE
Fix Unmounting Lazy Routes when Query Params Change

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Uses [the `render` prop of React Router's `<Route>`](https://reacttraining.com/react-router/web/api/Route/render-func) component in order to prevent full unmount/mount when query parameters change when using `<LazyRoute>`.

### CODE REVIEW

File: frontend/public/components/app.jsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ const DefaultPage = connectToFlags(FLAGS.OPENSHIFT)(({ flags }) => {
   return <Redirect to={statusPage} />;
 });
 
-const LazyRoute = (props) => <Route {...props} component={(componentProps) => <AsyncComponent loader={props.loader} kind={props.kind} {...componentProps} />} />;
+const LazyRoute = (props) => <Route {...props} render={(componentProps) => <AsyncComponent loader={props.loader} kind={props.kind} {...componentProps} />} />;
```
Comment (spadgett) [inline frontend/public/components/app.jsx]: 👍 

We should remove `catalogPageLoader` and `renderCatalogRoute` below

**Comment by spadgett** [comment]
cc @jeff-phillips-18

File: frontend/public/components/app.jsx (hunk 1 above)
Comment (alecmerdler) [reply frontend/public/components/app.jsx]: Done

File: frontend/public/components/app.jsx [hunk 2]
```diff
@@ -134,8 +120,8 @@ class App extends React.PureComponent {
             <LazyRoute path="/overview/ns/:ns" exact loader={() => import('./overview' /* webpackChunkName: "overview" */).then(m => m.OverviewPage)} />
             <Route path="/overview" exact component={NamespaceRedirect} />
 
-            {renderCatalogRoute('/catalog/all-namespaces')}
-            {renderCatalogRoute('/catalog/ns/:ns')}
+            <LazyRoute path="/catalog/all-namespaces" exact loader={import('./catalog/catalog-page').then(m => m.CatalogPage)} />
```
Comment (spadgett) [inline frontend/public/components/app.jsx]:
```suggestion
            <LazyRoute path="/catalog/all-namespaces" exact loader={() => import('./catalog/catalog-page').then(m => m.CatalogPage)} />
```

File: frontend/public/components/app.jsx [hunk 3]
```diff
@@ -134,8 +120,8 @@ class App extends React.PureComponent {
...
+            <LazyRoute path="/catalog/ns/:ns" exact loader={import('./catalog/catalog-page').then(m => m.CatalogPage)} />
```
Comment (spadgett) [inline frontend/public/components/app.jsx]:
```suggestion
            <LazyRoute path="/catalog/ns/:ns" exact loader={() => import('./catalog/catalog-page').then(m => m.CatalogPage)} />
```

File: frontend/public/components/app.jsx (hunk 2 above)
Comment (alecmerdler) [reply frontend/public/components/app.jsx]: Oh, whoops.

**Pull request merged**